rem/TimelineView.swift

```diff
@@ -29,96 +30,85 @@ struct TimelineView: View {
     
     var body: some View {
         ZStack {
-            let index = viewModel.currentFrameIndex
-            if let image = DatabaseManager.shared.getImage(index: index) {
-                let nsImage = NSImage(cgImage: image, size: NSSize(width: image.width, height: image.width))
-                CustomHostingControllerRepresentable(settingsManager: settingsManager, onClose: onClose, image: nsImage, analysis: $imageAnalysis, frame: frame)
-                    .frame(width: frame.width, height: frame.height)
-                    .ignoresSafeArea(.all)
-                    .onChange(of: viewModel.currentFrameIndex) {
-                        debounceTimer?.invalidate()
-                        debounceTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) { _ in
-                            if let image = DatabaseManager.shared.getImage(index: viewModel.currentFrameIndex) {
-                                analyzeImage(image)  // Assuming `image` is available here, modify as needed
-                            }
-                        }
-                    }
-                    .onAppear {
-                        debounceTimer?.invalidate()
-                        analyzeImage(image)
-                    }
-            } else {
```

Comment:
getting rid of this should fix a lot of issues due to all kinds of logic happening in `CustomHostingControllerRepresentable` like binding keys for quitting...